front/src/modules/settings/developers/utils.py/format-expiration.ts

```diff
@@ -0,0 +1,41 @@
+import { ApiFieldItem } from '@/settings/developers/types/ApiFieldItem';
+import { GetApiKeysQuery } from '~/generated/graphql';
+import { beautifyDateDiff } from '~/utils/date-utils';
+
+export const formatExpiration = (
+  expiresAt: string | null,
+  withExpiresMention: boolean = false,
+  short: boolean = true,
+) => {
+  let expiration = 'Never';
+  if (withExpiresMention) expiration = 'Never expires';
+  if (expiresAt) {
+    if (withExpiresMention) {
+      expiration = 'Expires in';
+    } else {
+      expiration = 'In';
+    }
+    expiration = `${expiration} ${beautifyDateDiff(
+      expiresAt,
+      undefined,
+      short,
+    )}`;
+  }
+  if (expiration.includes('-')) {
+    expiration = 'Expired';
+  }
+  return expiration;
```

Comment:
Maybe this would be simpler:

```suggestion
  if (expiresAt) {
    const dateDiff = beautifyDateDiff(
      expiresAt,
      undefined,
      short,
    );

    if (dateDiff.includes("-")) return "Expired";

    return withExpiresMention ? `Expires in ${dateDiff}` : `In ${dateDiff}`;
  }

  return withExpiresMention ? "Never expires" : "Never";
```